### PR TITLE
Metadata availability for implicit tiling

### PR DESCRIPTION
**Pull request by ErixenCruz**
Partially addresses https://github.com/CesiumGS/3d-tiles-next/issues/100

### CODE REVIEW

File: extensions/3DTILES_metadata/0.0.0/schema/3DTILES_metadata_implicit_tiling.schema.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "title": "3DTILES_metadata extension for implicit tiling subtree metadata availability.",
+  "type": "object",
+  "properties": {
+    "class": {
+      "description": "The class referring back to a class definition in 3DTILES_metadata in tileset.json",
+      "type": "string"
+    },
+    "additionalProperties": {
+      "$ref": "3DTILES_metadata_implicit_tiling_object.schema.json"
+    }
```
Comment (ptrgags) [inline extensions/3DTILES_metadata/0.0.0/schema/3DTILES_metadata_implicit_tiling.schema.json:12]: @lilleyse I talked with @ErixenCruz about this. we clarified the syntax for `allOf`, however, we had a question, so we didn't commit any changes yet.

Does this extension directly depend on `3DTILES_metadata`? if so, can we reuse the `instanceTable.property.schema.json` and just mention in the spec that JSON encoding is forbidden? 

If not, should we copy in the schema file and just remove the parts we don't use? Not sure if we should use the same filename or not.

File: extensions/3DTILES_metadata/0.0.0/schema/3DTILES_metadata_implicit_tiling_object.schema.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (ptrgags) [inline extensions/3DTILES_metadata/0.0.0/schema/3DTILES_metadata_implicit_tiling_object.schema.json:1]: Depending on the answer to the other question, we might want to merge this file with `instanceTable.property.schema.json`, just not sure which is the best way to do that yet.

**Review by ptrgags — COMMENTED** [review]
@ErixenCruz like we talked about, we need a little more clarification on this PR before I review again.

File: extensions/3DTILES_metadata/0.0.0/schema/3DTILES_metadata_implicit_tiling.schema.json (hunk 1 above)
Comment (lilleyse) [reply extensions/3DTILES_metadata/0.0.0/schema/3DTILES_metadata_implicit_tiling.schema.json:12]: Yeah it depends on `3DTILES_metadata`, and you can copy `instanceTable.property.schema.json` in the same way that was done for `EXT_feature_metadata` (I forget the specifics, but we would disallow JSON encoding here as well)

**Comment by lilleyse** [comment]
Ultimately I think this should depend on [3DTILES_tile_metadata](https://github.com/CesiumGS/3d-tiles/pull/429) instead of 3DTILES_metadata. However it does make the extension nestedness even worse which is a downside to splitting metadata across multiple extensions.

Maybe it's time to revisit 3DTILES_tile_metadata and get that into shape. Currently it just talks about explicit tiling. This branch would be a branch off of that.

File: extensions/3DTILES_metadata/0.0.0/schema/3DTILES_metadata_implicit_tiling.schema.json (hunk 1 above)
Comment (lilleyse) [reply extensions/3DTILES_metadata/0.0.0/schema/3DTILES_metadata_implicit_tiling.schema.json:12]: See https://github.com/CesiumGS/3d-tiles/pull/440#issuecomment-736117919. I think it actually depends on `3DTILES_tile_metadata`, which depends on `3DTILES_metadata`

**Comment by ErixenCruz** [comment]
What does getting 3DTILES_tile_metadata into shape entail? I can work on it. Is there an issue or list somewhere about what should be done?

**Comment by liuyingbin123** [comment]
If we can't get the actual size of tile in memory, we will never be able to accurately control the memory usage of tileset. Because the files on the disk are compressed.the browser will decompress  data in the process of using, which will lead to the increase of the amount of data.

**Comment by lilleyse** [comment]
Will be folded into [3DTILES_metadata](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/extensions/3DTILES_metadata)

**Pull request closed without merging**